python/mlcroissant/mlcroissant/_src/operation_graph/operations/download.py

```diff
@@ -128,6 +142,17 @@ class Download(Operation):
     def _download_from_http(self, filepath: epath.Path):
         response = requests.get(self.url, stream=True, timeout=10)
         total = int(response.headers.get("Content-Length", 0))
+
+        hash_type = _get_hash_type(self.node)
```

Comment:
This would then be simplified to `hash = _get_hash(self.node)`

Then you can use `hash.update(data)`